org.spoofax.interpreter.core/src/main/java/org/spoofax/interpreter/library/ssl/SSL_immutable_relation_intersect.java

```diff
@@ -0,0 +1,46 @@
+package org.spoofax.interpreter.library.ssl;
+
+import io.usethesource.capsule.BinaryRelation;
+
+import org.spoofax.interpreter.core.IContext;
+import org.spoofax.interpreter.library.AbstractPrimitive;
+import org.spoofax.interpreter.stratego.Strategy;
+import org.spoofax.interpreter.terms.IStrategoTerm;
+import java.util.Map;
+
+public class SSL_immutable_relation_intersect extends AbstractPrimitive {
+
+    protected SSL_immutable_relation_intersect() {
+        super("SSL_immutable_relation_intersect", 0, 1);
+    }
+
+    @Override
+    public boolean call(IContext env, Strategy[] sargs, IStrategoTerm[] targs) {
+        if(!(env.current() instanceof StrategoImmutableRelation)) {
+            return false;
+        }
+        if(!(targs[0] instanceof StrategoImmutableRelation)) {
+            return false;
```

Comment:
Failure vs exception. Not the current term, so again I'd consider this a programming error, not a reason for backtracking.